### PR TITLE
Avoid telemetry cluster metadata override

### PR DESCRIPTION
**Pull request by bianpengyuan**
Fixes https://github.com/istio/istio/issues/28140

telemetry metadata is overridden by configuration metadata.

### CODE REVIEW

File: pilot/pkg/networking/util/util.go [hunk 1]
```diff
@@ -385,18 +385,16 @@ func cloneLocalityLbEndpoints(endpoints []*endpoint.LocalityLbEndpoints) []*endp
...
-	AddConfigInfoMetadata(metadata, config)
-	return metadata
+	return AddConfigInfoMetadata(nil, config)
 }
 
 // AddConfigInfoMetadata adds name.namespace of the config, the type, etc
-// to the given core.Metadata struct.
-func AddConfigInfoMetadata(metadata *core.Metadata, config config.Meta) {
+// to the given core.Metadata struct, if metadata is not initialized, build a new metadata.
+func AddConfigInfoMetadata(metadata *core.Metadata, config config.Meta) *core.Metadata {
```
Comment (kyessenov) [inline pilot/pkg/networking/util/util.go:393]: You could do pointer-to-pointer here and not return.

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/networking/util/util.go (hunk 1 above)
Comment (bianpengyuan) [reply pilot/pkg/networking/util/util.go:393]: I tried that out, but does not seem to be intuitive comparing to the current code.

Comment (kyessenov) [reply pilot/pkg/networking/util/util.go:393]: Yeah, I figured it. Requires some C++ mindset :)

**Comment by bianpengyuan** [comment]
@istio/wg-policies-and-telemetry-maintainers ptal, needs approval on this. Thanks!

**Review by kyessenov — APPROVED** [review]

**Comment by bianpengyuan** [comment]
Test failure was due to change of inbound cluster name. Merge upstream and retest..

**Pull request merged**